packages/flutter_tools/lib/src/observatory.dart

```diff
@@ -114,17 +114,14 @@ class Observatory {
     });
   }
 
-  Future<Response> reloadSources(String isolateId) async {
-    Completer<Event> whenIsolateReloads = new Completer<Event>();
-    StreamSubscription<Event> sub = onIsolateEvent
-      .where((Event event) => event.kind == 'IsolateReload')
-      .listen((Event event) => whenIsolateReloads.complete(event));
-
+  // Returns null on success and the reload failure reason on failure.
+  Future<String> reloadSources(String isolateId) async {
```

Comment:
Currently this doesn't return until the VM send the `IsolateReload` event - this is by design so that we can get a good sense of how long the reload actually takes. Has the behavior of the VM changed? If we more that behavior (waiting for the event) out of this call, it would be nice to expose it elsewhere.